Deleting the Events content type causes system errors

Version 5.3.8.2 
Reported in https://dotcms.zendesk.com/agent/tickets/103686
Recreated locally.

**Describe the bug**
Deleting the Events content type causes system errors.

**To Reproduce**
Steps to reproduce the behavior:
1. Install fresh 5.3.8.2 instance
2. Delete the Events content type (variable calendarEvents)
3. go to Content->Search
4. See error

ERROR IN CONTENT->SEARCH
```

An unexpected system error occurred.

An exception occurred processing JSP page [/html/portal/view_portlet_inc.jsp] at line [87] 84: 85: boolean portletException = false; 86: 87: cachePortlet.render(renderRequest, renderResponse); 88: 89: SessionMessages.clear(renderRequest); 90: SessionErrors.clear(renderRequest); Stacktrace:
```


![image](https://user-images.githubusercontent.com/49727631/109556631-b11eba80-7a9c-11eb-857d-480a2b21d93b.png)


STACK TRACE:

```
[01/03/21 20:31:18:662 GMT] ERROR portlet.PortletRequestDispatcherImpl: An exception occurred processing JSP page [/html/portal/layout_portal.jsp] at line [19]

16: <% if (Validator.isNotNull(tilesPortletSubNav) ) {%>
17: <liferay:include page="<%= Constants.TEXT_HTML_DIR + tilesPortletSubNav %>" flush="true" />
18: <%}%>
19: <jsp:include page="<%= Constants.TEXT_HTML_DIR + tilesContent %>"></jsp:include>
20: </div>
21:
22: <%}else{ %>


Stacktrace:
org.apache.jasper.JasperException: An exception occurred processing JSP page [/html/portal/layout_portal.jsp] at line [19]

16: <% if (Validator.isNotNull(tilesPortletSubNav) ) {%>
17: <liferay:include page="<%= Constants.TEXT_HTML_DIR + tilesPortletSubNav %>" flush="true" />
18: <%}%>
19: <jsp:include page="<%= Constants.TEXT_HTML_DIR + tilesContent %>"></jsp:include>
20: </div>
21:
22: <%}else{ %>


Stacktrace:
at org.apache.jasper.servlet.JspServletWrapper.handleJspException(JspServletWrapper.java:584) ~[jasper.jar:8.5.32]
at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:466) ~[jasper.jar:8.5.32]
at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:386) ~[jasper.jar:8.5.32]
at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:330) ~[jasper.jar:8.5.32]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:728) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.doInclude(ApplicationDispatcher.java:591) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.include(ApplicationDispatcher.java:527) ~[catalina.jar:8.5.32]
at com.liferay.portlet.PortletRequestDispatcherImpl.include(PortletRequestDispatcherImpl.java:206) ~[dotcms_5.3.8.2_999999.jar:?]
at com.liferay.portal.struts.PortletRequestProcessor.doInclude(PortletRequestProcessor.java:339) ~[dotcms_5.3.8.2_999999.jar:?]
at com.liferay.portal.struts.PortletRequestProcessor.doForward(PortletRequestProcessor.java:314) ~[dotcms_5.3.8.2_999999.jar:?]
at com.dotcms.repackage.org.apache.struts.tiles.TilesRequestProcessor.processTilesDefinition(TilesRequestProcessor.java:239) ~[dot.struts-1.2.10_1.jar:?]
at com.dotcms.repackage.org.apache.struts.tiles.TilesRequestProcessor.processForwardConfig(TilesRequestProcessor.java:302) ~[dot.struts-1.2.10_1.jar:?]
at com.dotcms.repackage.com.oroad.stxx.plugin.StxxTilesRequestProcessor.processForwardConfig(StxxTilesRequestProcessor.java:134) ~[dot.stxx-1.3_3.jar:?]
at com.liferay.portal.struts.PortletRequestProcessor.processForwardConfig(PortletRequestProcessor.java:201) ~[dotcms_5.3.8.2_999999.jar:?]
at com.dotcms.repackage.org.apache.struts.action.RequestProcessor.process(RequestProcessor.java:241) ~[dot.struts-1.2.10_1.jar:?]
at com.liferay.portal.struts.PortletRequestProcessor.process(PortletRequestProcessor.java:87) ~[dotcms_5.3.8.2_999999.jar:?]
at com.liferay.portlet.StrutsPortlet.include(StrutsPortlet.java:152) ~[dotcms_5.3.8.2_999999.jar:?]
at com.liferay.portlet.StrutsPortlet.doView(StrutsPortlet.java:104) ~[dotcms_5.3.8.2_999999.jar:?]
at com.dotcms.repackage.javax.portlet.GenericPortlet.doDispatch(GenericPortlet.java:235) ~[dot.portlet-1.0_2.jar:?]
at com.dotcms.repackage.javax.portlet.GenericPortlet.render(GenericPortlet.java:163) ~[dot.portlet-1.0_2.jar:?]
at com.liferay.portlet.ConcretePortletWrapper.render(ConcretePortletWrapper.java:114) ~[dotcms_5.3.8.2_999999.jar:?]
at org.apache.jsp.html.portal.layout_jsp._jspService(layout_jsp.java:625) ~[?:?]
at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70) ~[jasper.jar:8.5.32]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:443) ~[jasper.jar:8.5.32]
at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:386) ~[jasper.jar:8.5.32]
at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:330) ~[jasper.jar:8.5.32]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:728) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.doInclude(ApplicationDispatcher.java:591) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.include(ApplicationDispatcher.java:527) ~[catalina.jar:8.5.32]
at org.apache.jasper.runtime.JspRuntimeLibrary.include(JspRuntimeLibrary.java:892) ~[jasper.jar:8.5.32]
at org.apache.jsp.html.portal.layout_005fportal_jsp._jspService(layout_005fportal_jsp.java:1583) ~[?:?]
at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70) ~[jasper.jar:8.5.32]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:443) ~[jasper.jar:8.5.32]
at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:386) ~[jasper.jar:8.5.32]
at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:330) ~[jasper.jar:8.5.32]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) ~[tomcat-websocket.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:728) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:470) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:395) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:316) ~[catalina.jar:8.5.32]
at com.liferay.portal.struts.StrutsUtil.forward(StrutsUtil.java:72) ~[dotcms_5.3.8.2_999999.jar:?]
at com.liferay.portal.struts.PortalRequestProcessor.doForward(PortalRequestProcessor.java:161) ~[dotcms_5.3.8.2_999999.jar:?]
at com.dotmarketing.struts.PortalRequestProcessor.doForward(PortalRequestProcessor.java:48) ~[dotcms_5.3.8.2_999999.jar:?]
at com.dotcms.repackage.org.apache.struts.tiles.TilesRequestProcessor.processTilesDefinition(TilesRequestProcessor.java:239) ~[dot.struts-1.2.10_1.jar:?]
at com.dotcms.repackage.org.apache.struts.tiles.TilesRequestProcessor.processForwardConfig(TilesRequestProcessor.java:302) ~[dot.struts-1.2.10_1.jar:?]
at com.dotcms.repackage.com.oroad.stxx.plugin.StxxTilesRequestProcessor.processForwardConfig(StxxTilesRequestProcessor.java:134) ~[dot.stxx-1.3_3.jar:?]
at com.dotcms.repackage.org.apache.struts.action.RequestProcessor.process(RequestProcessor.java:241) ~[dot.struts-1.2.10_1.jar:?]
at com.liferay.portal.struts.PortalRequestProcessor.process(PortalRequestProcessor.java:147) ~[dotcms_5.3.8.2_999999.jar:?]
at com.dotcms.repackage.org.apache.struts.action.ActionServlet.process(ActionServlet.java:1202) ~[dot.struts-1.2.10_1.jar:?]
at com.dotcms.repackage.org.apache.struts.action.ActionServlet.doGet(ActionServlet.java:412) ~[dot.struts-1.2.10_1.jar:?]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:635) ~[servlet-api.jar:?]
at com.liferay.portal.servlet.MainServlet.callParentService(MainServlet.java:249) ~[dotcms_5.3.8.2_999999.jar:?]
at com.liferay.portal.servlet.MainServlet.service(MainServlet.java:449) ~[dotcms_5.3.8.2_999999.jar:?]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) ~[tomcat-websocket.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at com.dotcms.repackage.com.liferay.filters.secure.SecureFilter.doFilter(SecureFilter.java:135) ~[dot.secure-filter-ukv_2.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at com.dotmarketing.filters.CMSFilter.doFilterInternal(CMSFilter.java:198) ~[dotcms_5.3.8.2_999999.jar:?]
at com.dotmarketing.filters.CMSFilter.doFilter(CMSFilter.java:48) ~[dotcms_5.3.8.2_999999.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[dotcms_5.3.8.2_999999.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[dotcms_5.3.8.2_999999.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at com.dotcms.vanityurl.filters.VanityURLFilter.doFilter(VanityURLFilter.java:91) ~[dotcms_5.3.8.2_999999.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at org.tuckey.web.filters.urlrewrite.RuleChain.handleRewrite(RuleChain.java:176) ~[urlrewritefilter-4.0.4.jar:4.0.4]
at org.tuckey.web.filters.urlrewrite.RuleChain.doRules(RuleChain.java:145) ~[urlrewritefilter-4.0.4.jar:4.0.4]
at org.tuckey.web.filters.urlrewrite.UrlRewriter.processRequest(UrlRewriter.java:92) ~[urlrewritefilter-4.0.4.jar:4.0.4]
at org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:389) ~[urlrewritefilter-4.0.4.jar:4.0.4]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at com.dotmarketing.filters.TimeMachineFilter.doFilter(TimeMachineFilter.java:134) ~[dotcms_5.3.8.2_999999.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at com.dotmarketing.filters.ThreadNameFilter.doFilter(ThreadNameFilter.java:88) ~[dotcms_5.3.8.2_999999.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at com.dotmarketing.filters.CookiesFilter.doFilter(CookiesFilter.java:48) ~[dotcms_5.3.8.2_999999.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at com.dotmarketing.filters.CharsetEncodingFilter.doFilter(CharsetEncodingFilter.java:99) ~[dotcms_5.3.8.2_999999.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[dotcms_5.3.8.2_999999.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at com.dotcms.filters.NormalizationFilter.doFilter(NormalizationFilter.java:73) ~[dotcms_5.3.8.2_999999.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96) ~[catalina.jar:8.5.32]
at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140) ~[catalina.jar:8.5.32]
at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81) ~[catalina.jar:8.5.32]
at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:650) ~[catalina.jar:8.5.32]
at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:685) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87) ~[catalina.jar:8.5.32]
at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342) ~[catalina.jar:8.5.32]
at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:800) ~[tomcat-coyote.jar:8.5.32]
at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66) ~[tomcat-coyote.jar:8.5.32]
at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:800) ~[tomcat-coyote.jar:8.5.32]
at org.apache.tomcat.util.net.Nio2Endpoint$SocketProcessor.doRun(Nio2Endpoint.java:1699) ~[tomcat-coyote.jar:8.5.32]
at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) ~[tomcat-coyote.jar:8.5.32]
at org.apache.tomcat.util.net.AbstractEndpoint.processSocket(AbstractEndpoint.java:1050) ~[tomcat-coyote.jar:8.5.32]
at org.apache.tomcat.util.net.Nio2Endpoint$Nio2SocketWrapper$4.completed(Nio2Endpoint.java:630) ~[tomcat-coyote.jar:8.5.32]
at org.apache.tomcat.util.net.Nio2Endpoint$Nio2SocketWrapper$4.completed(Nio2Endpoint.java:608) ~[tomcat-coyote.jar:8.5.32]
at sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:126) ~[?:1.8.0_212]
at sun.nio.ch.UnixAsynchronousSocketChannelImpl.finishRead(UnixAsynchronousSocketChannelImpl.java:430) ~[?:1.8.0_212]
at sun.nio.ch.UnixAsynchronousSocketChannelImpl.finish(UnixAsynchronousSocketChannelImpl.java:191) ~[?:1.8.0_212]
at sun.nio.ch.UnixAsynchronousSocketChannelImpl.onEvent(UnixAsynchronousSocketChannelImpl.java:213) ~[?:1.8.0_212]
at sun.nio.ch.EPollPort$EventHandlerTask.run(EPollPort.java:293) ~[?:1.8.0_212]
at sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:112) ~[?:1.8.0_212]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_212]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_212]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
Caused by: org.apache.jasper.JasperException: An exception occurred processing JSP page [/html/portlet/ext/contentlet/view_contentlets_js_inc.jsp] at line [18]

15:
16: <%boolean canReindex= APILocator.getRoleAPI().doesUserHaveRole(user,APILocator.getRoleAPI().loadRoleByKey(Role.CMS_POWER_USER))|| com.dotmarketing.business.APILocator.getRoleAPI().doesUserHaveRole(user,com.dotmarketing.business.APILocator.getRoleAPI().loadCMSAdminRole());%>
17:
18: <%Structure calendarEventSt = APILocator.getStructureAPI().findByVarName("calendarEvent", APILocator.getUserAPI().getSystemUser());%>
19:
20: dojo.require("dojox.dtl.filter.strings");
21: dojo.require("dijit.form.FilteringSelect");


Stacktrace:
at org.apache.jasper.servlet.JspServletWrapper.handleJspException(JspServletWrapper.java:584) ~[jasper.jar:8.5.32]
at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:466) ~[jasper.jar:8.5.32]
at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:386) ~[jasper.jar:8.5.32]
at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:330) ~[jasper.jar:8.5.32]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:728) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.doInclude(ApplicationDispatcher.java:591) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.include(ApplicationDispatcher.java:527) ~[catalina.jar:8.5.32]
at org.apache.jasper.runtime.JspRuntimeLibrary.include(JspRuntimeLibrary.java:892) ~[jasper.jar:8.5.32]
at org.apache.jsp.html.portal.layout_005fportal_jsp._jspService(layout_005fportal_jsp.java:630) ~[?:?]
at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70) ~[jasper.jar:8.5.32]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:443) ~[jasper.jar:8.5.32]
... 136 more
Caused by: javax.servlet.ServletException: com.dotcms.contenttype.exception.NotFoundInDbException: Content Type with id:'calendarEvent' not found
at org.apache.jasper.runtime.PageContextImpl.doHandlePageException(PageContextImpl.java:909) ~[jasper.jar:8.5.32]
at org.apache.jasper.runtime.PageContextImpl.handlePageException(PageContextImpl.java:838) ~[jasper.jar:8.5.32]
at org.apache.jsp.html.portlet.ext.contentlet.view_005fcontentlets_jsp._jspService(view_005fcontentlets_jsp.java:5820) ~[?:?]
at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70) ~[jasper.jar:8.5.32]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:443) ~[jasper.jar:8.5.32]
at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:386) ~[jasper.jar:8.5.32]
at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:330) ~[jasper.jar:8.5.32]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:728) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.doInclude(ApplicationDispatcher.java:591) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.include(ApplicationDispatcher.java:527) ~[catalina.jar:8.5.32]
at org.apache.jasper.runtime.JspRuntimeLibrary.include(JspRuntimeLibrary.java:892) ~[jasper.jar:8.5.32]
at org.apache.jsp.html.portal.layout_005fportal_jsp._jspService(layout_005fportal_jsp.java:630) ~[?:?]
at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70) ~[jasper.jar:8.5.32]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:443) ~[jasper.jar:8.5.32]
... 136 more
Caused by: com.dotcms.contenttype.exception.NotFoundInDbException: Content Type with id:'calendarEvent' not found
at com.dotcms.contenttype.business.ContentTypeFactoryImpl.dbById(ContentTypeFactoryImpl.java:281) ~[dotcms_5.3.8.2_999999.jar:?]
at com.dotcms.contenttype.business.ContentTypeFactoryImpl.find(ContentTypeFactoryImpl.java:108) ~[dotcms_5.3.8.2_999999.jar:?]
at com.dotcms.contenttype.business.ContentTypeAPIImpl.find_aroundBody2(ContentTypeAPIImpl.java:112) ~[dotcms_5.3.8.2_999999.jar:?]
at com.dotcms.contenttype.business.ContentTypeAPIImpl$AjcClosure3.run(ContentTypeAPIImpl.java:1) ~[dotcms_5.3.8.2_999999.jar:?]
at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_5.3.8.2_999999.jar:?]
at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_5.3.8.2_999999.jar:?]
at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_5.3.8.2_999999.jar:?]
at com.dotcms.contenttype.business.ContentTypeAPIImpl.find(ContentTypeAPIImpl.java:108) ~[dotcms_5.3.8.2_999999.jar:?]
at com.dotmarketing.portlets.structure.business.StructureAPIImpl.findByVarName(StructureAPIImpl.java:181) ~[dotcms_5.3.8.2_999999.jar:?]
at org.apache.jsp.html.portlet.ext.contentlet.view_005fcontentlets_jsp._jspService(view_005fcontentlets_jsp.java:1247) ~[?:?]
at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70) ~[jasper.jar:8.5.32]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:443) ~[jasper.jar:8.5.32]
at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:386) ~[jasper.jar:8.5.32]
at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:330) ~[jasper.jar:8.5.32]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:728) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.doInclude(ApplicationDispatcher.java:591) ~[catalina.jar:8.5.32]
at org.apache.catalina.core.ApplicationDispatcher.include(ApplicationDispatcher.java:527) ~[catalina.jar:8.5.32]
at org.apache.jasper.runtime.JspRuntimeLibrary.include(JspRuntimeLibrary.java:892) ~[jasper.jar:8.5.32]
at org.apache.jsp.html.portal.layout_005fportal_jsp._jspService(layout_005fportal_jsp.java:630) ~[?:?]
at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70) ~[jasper.jar:8.5.32]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:?]
at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:443) ~[jasper.jar:8.5.32]
... 136 more
[01/03/21 20:31:18:665 GMT] ERROR portlet.StrutsPortlet: org.apache.jasper.JasperException: An exception occurred processing JSP page [/html/portlet/ext/contentlet/view_contentlets_js_inc.jsp] at line [18]

15:
16: <%boolean canReindex= APILocator.getRoleAPI().doesUserHaveRole(user,APILocator.getRoleAPI().loadRoleByKey(Role.CMS_POWER_USER))|| com.dotmarketing.business.APILocator.getRoleAPI().doesUserHaveRole(user,com.dotmarketing.business.APILocator.getRoleAPI().loadCMSAdminRole());%>
17:
18: <%Structure calendarEventSt = APILocator.getStructureAPI().findByVarName("calendarEvent", APILocator.getUserAPI().getSystemUser());%>
19:
20: dojo.require("dojox.dtl.filter.strings");
21: dojo.require("dijit.form.FilteringSelect");


Stacktrace:
[01/03/21 20:31:19:007 GMT] ERROR servlet.MainServlet: An exception occurred processing JSP page [/html/portal/view_portlet_inc.jsp] at line [87]

84:
85: boolean portletException = false;
86:
87: cachePortlet.render(renderRequest, renderResponse);
88:
89: SessionMessages.clear(renderRequest);
90: SessionErrors.clear(renderRequest);


Stacktrace:
```

--- a/dotCMS/src/main/webapp/html/portlet/ext/contentlet/view_contentlets_js_inc.jsp
+++ b/dotCMS/src/main/webapp/html/portlet/ext/contentlet/view_contentlets_js_inc.jsp
@@ -4,10 +4,12 @@
 <%@page import="com.dotmarketing.business.APILocator"%>
 <%@page import="com.dotmarketing.business.PermissionAPI"%>
 <%@page import="com.dotmarketing.business.Role"%>
-<%@page import="com.dotmarketing.portlets.structure.model.Structure"%>
+<%@page import="com.dotcms.contenttype.model.type.ContentType"%>
 <%@page import="com.dotmarketing.util.UtilMethods"%>
 <%@page import="com.liferay.portal.language.LanguageUtil"%>
+<%@page import="com.dotmarketing.portlets.structure.model.Structure"%>
 <%@page import="java.util.List"%>
+<%@ page import="io.vavr.control.Try" %>
 
 
 <script language="JavaScript"><!--
@@ -15,7 +17,10 @@
 
 <%boolean canReindex= APILocator.getRoleAPI().doesUserHaveRole(user,APILocator.getRoleAPI().loadRoleByKey(Role.CMS_POWER_USER))|| com.dotmarketing.business.APILocator.getRoleAPI().doesUserHaveRole(user,com.dotmarketing.business.APILocator.getRoleAPI().loadCMSAdminRole());%>
 
-<%Structure calendarEventSt = APILocator.getStructureAPI().findByVarName("calendarEvent", APILocator.getUserAPI().getSystemUser());%>
+<%
+final ContentType calendarEventSt = Try.of(()->APILocator.getContentTypeAPI(APILocator.systemUser()).find("calendarEvent")).getOrNull();
+final String calendarEventInode = null!=calendarEventSt ? calendarEventSt.inode() : StringPool.BLANK;
+%>
 
         dojo.require("dojox.dtl.filter.strings");
         dojo.require("dijit.form.FilteringSelect");
@@ -330,7 +335,7 @@
 
                 var editRef ='';
 
-                if(structure_id == '<%=calendarEventSt.getInode() %>'){
+                if(structure_id == '<%=calendarEventInode %>'){
               editRef = " editEvent('" + inode + "','<%=user.getUserId()%>','<%= referer %>'," + liveSt + "," + workingSt + "," + write + ") ";
             }else{
               editRef = " editContentlet('" + inode + "','<%=user.getUserId()%>','<%= referer %>'," + liveSt + "," + workingSt + "," + write + ") ";
@@ -386,7 +391,7 @@
 
                 var editRef = '';
 
-            if(structure_id == '<%=calendarEventSt.getInode() %>'){
+            if(structure_id == '<%=calendarEventInode %>'){
               editRef = " editEvent('" + inode + "','<%=user.getUserId()%>','<%= referer %>'," + liveSt + "," + workingSt + "," + write + ") ";
             }else{
               editRef = " editContentlet('" + inode + "','<%=user.getUserId()%>','<%= referer %>'," + liveSt + "," + workingSt + "," + write + ") ";
@@ -844,7 +849,7 @@
 				dijit.byId("selectStructureDiv").show();
 				return;
 			}
-          else if(structureInode == '<%=calendarEventSt.getInode() %>'){
+          else if(structureInode == '<%=calendarEventInode %>'){
                 var href = "<portlet:actionURL windowState='<%= WindowState.MAXIMIZED.toString() %>'>";
                 href += "<portlet:param name='struts_action' value='/ext/calendar/edit_event' />";
                 href += "<portlet:param name='cmd' value='new' />";
@@ -1525,7 +1530,7 @@
 
                                             }
                                           }
-                                
+
                                 }else {
                                         fieldsValues[fieldsValues.length] = selectedStruct+"."+field["fieldVelocityVarName"];
                                         fieldsValues[fieldsValues.length] = formField.value;
@@ -1700,7 +1705,7 @@
                 if (viewCard) {
 	                viewCard.value = '';
                 }
-                
+
                 for (var i = 0; i < cbContentInodeList.length; ++i) {
                         check = dijit.byId("checkbox" + i);
                         if(check) {
@@ -1862,7 +1867,7 @@
 			var workingSt = data.working === "true" ? "1" : "0";
 			var write = userHasWritePermission (data, userId) ? "1" : "0";
 
-			if (data.structureInode == '<%=calendarEventSt.getInode() %>') {
+			if (data.structureInode == '<%=calendarEventInode %>') {
 				editEvent(inode, '<%=user.getUserId()%>', '<%= referer %>', liveSt, workingSt, write);
 			}else{
 				editContentlet(inode, '<%=user.getUserId()%>', '<%= referer %>', liveSt, workingSt, write);
@@ -1890,7 +1895,7 @@
 			const wfActionMapList = JSON.parse(data["wfActionMapList"]);
 
 			if ((live || working) && (read=="1") && (!deleted)) {
-				if(structure_id == '<%=calendarEventSt.getInode() %>'){
+				if(structure_id == '<%=calendarEventInode %>'){
 					actions.push({ label: write === '1' ? '<%=LanguageUtil.get(pageContext, "Edit") %>' : '<%=LanguageUtil.get(pageContext, "View") %>',
 						action: () => { editEvent(data.inode, '<%= user.getUserId() %>', '<%= referer %>', liveSt, workingSt, write)}
 					});
@@ -1917,7 +1922,7 @@
 			}
 
 			if (locked && (write=="1")){
-				if(structure_id == '<%=calendarEventSt.getInode() %>') {
+				if(structure_id == '<%=calendarEventInode %>') {
 					actions.push({ label: '<%=LanguageUtil.get(pageContext, "Unlock") %>',
 						action: () => { unlockEvent(data.inode, '<%= user.getUserId() %>', '<%= referer %>', liveSt, workingSt, write)}
 					});
@@ -2037,10 +2042,10 @@
                         cellData.contentTypeIcon : getIconName(cellData['__type__']);
                     var hasTitleImage = (cellData.hasTitleImage ==='true');
 
-                    cell.innerHTML = (hasTitleImage) 
-                        ? '<img draggable="false" style="width:64px;height: 64px;object-fit: contain;" class="listingTitleImg" onError="replaceWithIcon(this.parentElement, \'' + iconName + '\')" src="/dA/' + cellData.inode + '/titleImage/256w" alt="' + cellData['__title__'].replace(/[^A-Za-z0-9_]/g, ' ') + '" >' 
+                    cell.innerHTML = (hasTitleImage)
+                        ? '<img draggable="false" style="width:64px;height: 64px;object-fit: contain;" class="listingTitleImg" onError="replaceWithIcon(this.parentElement, \'' + iconName + '\')" src="/dA/' + cellData.inode + '/titleImage/256w" alt="' + cellData['__title__'].replace(/[^A-Za-z0-9_]/g, ' ') + '" >'
                         : '<dot-contentlet-icon icon="' + iconName +'" size="48px" />';
-                    
+
                     cell.setAttribute("style","height: 85px; text-align: center;");
 
                     for (var j = 0; j < headers.length; j++) {
@@ -2054,7 +2059,7 @@
                             let fieldVarTitle = cellData[fieldVarName + "_title_"];
                             fieldVarTitle     = fieldVarTitle || cellData[fieldVarName]
                             var value         = titleCell(cellData,fieldVarTitle, i);
-                            
+
                             if (value != null){
                                 cell.innerHTML = value;
                             }
@@ -2081,7 +2086,7 @@
                             let fieldVarTitle = cellData[fieldVarName + "_title_"];
                             fieldVarTitle     = fieldVarTitle || cellData[fieldVarName]
                             var value         = fieldVarTitle;
-                        
+
                             if (value != null){
                                 cell.innerHTML = value;
                             }
@@ -2138,7 +2143,7 @@
                     // NEW CONTEXT MENU
 
                     if ((live || working) && (read=="1") && (!deleted)) {
-                            if(structure_id == '<%=calendarEventSt.getInode() %>'){
+                            if(structure_id == '<%=calendarEventInode %>'){
                                 if (write=="1"){
                                 popupMenuItems += "<div dojoType=\"dijit.MenuItem\" iconClass=\"editIcon\" onClick=\"editEvent('" + cellData.inode + "','<%= user.getUserId() %>','<%= referer %>'," + liveSt + "," + workingSt + "," + write + ");\"><%=LanguageUtil.get(pageContext, "Edit") %></div>";
                                 }else{
@@ -2179,7 +2184,7 @@
                     // END NEW CONTEXT
 
                     if (locked && (write=="1")){
-                        if(structure_id == '<%=calendarEventSt.getInode() %>'){
+                        if(structure_id == '<%=calendarEventInode %>'){
                             popupMenuItems += "<div dojoType=\"dijit.MenuItem\" iconClass=\"unlockIcon\" onClick=\"unlockEvent('" + cellData.inode + "','<%= user.getUserId() %>','<%= referer %>'," + liveSt + "," + workingSt + "," + write + ");\"><%=LanguageUtil.get(pageContext, "Unlock") %></div>";
                         }else{
                             popupMenuItems += "<div dojoType=\"dijit.MenuItem\" iconClass=\"unlockIcon\" onClick=\"_unlockAsset('" + cellData.inode + "');\"><%=LanguageUtil.get(pageContext, "Unlock") %></div>";

